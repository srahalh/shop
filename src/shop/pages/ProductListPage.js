import React from "react";
import { useGetProductsQuery } from "../redux/API/productsAPI";


export const ProductListPage = () => {

  const { data: products} = useGetProductsQuery();

  return (
    <>
      {
        console.log(products)
        //el boton de atras se hace con hooks de useNavigate (mirar https://www.udemy.com/course/react-cero-experto/learn/lecture/19944240#overview)
      }
      <div>ProductListPage</div>
    </>
  );
};
