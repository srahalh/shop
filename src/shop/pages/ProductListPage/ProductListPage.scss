@import '../../styles/mixins.scss';
@import '../../styles/config.scss';

.ProductListPage__container {
  @include container-tablet;
  
    @include desktop {
      @include container-desktop;
    }
  
    @include desktop-lg {
      @include container-desktop-lg;
    }
    
  .ProductListPage__List {
    @include flex-layout-columns;
    margin-top: $margin-md;
  
    @include desktop {
      @include flex-layout-space-start;
      margin-top: $margin-md;
    }
  
    @include desktop-lg {
      @include flex-layout-space-start;
      margin-top: $margin-md;
    }
    flex-wrap: wrap;
  
    .ProductListPage__search{
      flex: 1;
    }
    .ProductListPage__emptyList {
      color: $color-error;
    }
  }
  
}