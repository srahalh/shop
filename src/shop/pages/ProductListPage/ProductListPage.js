import React from 'react';
import ErrorMessage from '../../components/ErrorMessage/ErrorMessage';
import { MESSAGES } from '../../components/ErrorMessage/Messages';
import ItemCard from '../../components/Item/ItemCard/ItemCard';
import Search from '../../components/Search/Search';
import Spinner from '../../components/Spinner/Spinner';
import useSearch from '../../hooks/useSearch';
import { useGetProductsQuery } from '../../redux/API/productsAPI';

import './ProductListPage.scss';

export const ProductListPage = () => {
  const { data: products = [], isLoading, isError } = useGetProductsQuery();
  const { elementsSearched: productSearched, setSearchValue } =
    useSearch(products);

  return (
    <div className="ProductListPage__container">
      <Search setSearchValue={setSearchValue} />
      <div className="ProductListPage__List">
        {isLoading ? (
          <Spinner />
        ) : productSearched.length === 0 || isError ? (
          <ErrorMessage message={MESSAGES.PRODUCT_LIST_PAGE_MESSAGE} />
        ) : (
          productSearched.map((product, index) => {
            return (
              <ItemCard
                key={index}
                id={product.id}
                model={product.model}
                brand={product.brand}
                price={product.price}
                imageURL={product.imgUrl}
              />
            );
          })
        )}
      </div>
    </div>
  );
};
