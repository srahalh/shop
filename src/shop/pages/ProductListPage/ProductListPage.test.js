import React from "react";
import { ProductListPage } from "./ProductListPage";
import { renderWithProviders } from "../../test/setupTests";
const fulfilledState = require("../../test/fulfilledPoductsState.json");


describe("Test ProductListPage", () => {
  test("Should render ProductListPage", () => {
    const { asFragment  } = renderWithProviders(<ProductListPage />, { preloadedState: fulfilledState });
    expect(asFragment ()).toMatchSnapshot();
  });
});
