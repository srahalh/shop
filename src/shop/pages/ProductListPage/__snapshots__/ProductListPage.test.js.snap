// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductListPage Should render ProductListPage 1`] = `
<DocumentFragment>
  <div
    class="ProductListPage__container"
  >
    <div
      class="Search__container"
    >
      <input
        label="search"
        placeholder="Search article by category or title"
        type="text"
      />
    </div>
    <div
      class="ProductListPage__List"
    >
      <article
        class="ItemCard__container"
      >
        <img
          alt="Some images here"
          class="ItemImage__container"
          loading="lazy"
          src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
        />
        <div
          class="ItemCard__info"
        >
          <h3>
            Liquid 1
          </h3>
          <p>
            Brand: Acer
          </p>
          <p>
            Price:
            <strong>
               16 €
            </strong>
          </p>
        </div>
      </article>
      <article
        class="ItemCard__container"
      >
        <img
          alt="Some images here"
          class="ItemImage__container"
          loading="lazy"
          src="https://fakestoreapi.com/img/61U7T1koQqL._AC_SX679_.jpg"
        />
        <div
          class="ItemCard__info"
        >
          <h3>
            Liquid 2
          </h3>
          <p>
            Brand: Acer
          </p>
          <p>
            Price:
            <strong>
               16 €
            </strong>
          </p>
        </div>
      </article>
    </div>
  </div>
</DocumentFragment>
`;
